src/timeMachine/components/view_options/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,52 @@
+import React, {FC, useState} from 'react'
+import {useSelector} from 'react-redux'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+
+import {getVisTable} from 'src/timeMachine/selectors'
+
+import {useVisXDomainSettings} from 'src/shared/utils/useVisDomainSettings'
+
+interface Props {
+  setDomain: (xDomain: [number, number]) => void
+  xDomain: number[] | null
+}
+
+const isValidXDomain = xDomain => {
+  return (
+    xDomain &&
+    xDomain.length === 2 &&
+    typeof xDomain[0] === 'number' &&
+    typeof xDomain[1] === 'number'
+  )
+}
+
+export const TimeDomainAutoToggle: FC<Props> = ({setDomain, xDomain}) => {
+  const [isActive, setIsActive] = useState<boolean>(false)
+
+  const {table} = useSelector(getVisTable)
+
+  const [xDom] = useVisXDomainSettings(
+    xDomain,
+    table.getColumn('_time', 'number')
+  )
+
+  const toggleActiveDomain = (): void => {
+    if (!isActive) {
+      setDomain([xDom[0], xDom[1]])
+    } else {
+      setDomain(null)
+    }
+    setIsActive(!isActive)
+  }
+  return (
+    <div data-testid="time-domain-toggle">
+      <SlideToggle
+        size={ComponentSize.ExtraSmall}
+        active={isActive}
+        onChange={toggleActiveDomain}
+        disabled={!isValidXDomain(xDom)}
```

Comment:
is this needed here? can we instead just check if `props.xDomain` exists, and if it doesn't, just skip all this process and render this disabled?

under what conditions is a good `xDomain` passed in, but a bad `xDom` (or whatever it's called) is produced?

Reply:
Because this allows us not to worry about what props.xDomain is. Whatever is happening outside this component, we now wont mess up the graph if somehow a bad value does make it in.